state/factory/factory.go

```diff
@@ -581,7 +574,7 @@ func (sf *factory) stateAtHeight(height uint64, ns string, key []byte, s interfa
 	if !sf.saveHistory {
 		return ErrNoArchiveData
 	}
-	tlt, err := newTwoLayerTrie(fmt.Sprintf("%s-%d", ArchiveTrieRootKey, height), sf.dao, false)
+	tlt, err := newTwoLayerTrie(ArchiveTrieNamespace, sf.dao, fmt.Sprintf("%s-%d", ArchiveTrieRootKey, height), false)
```

Comment:
nit:
feel it is more natural to use []byte
rootKey = append([]byte(ArchiveTrieRootKey), uint64tobytes(height))